snippets/goobox.md

```diff
@@ -0,0 +1,42 @@
+---
+title: goobox
+tags: image hosting, regex, url, google drive, dropbox, advanced
+---
+
+- string, Returns a string url that can be used directly as source of an image.
+- when you host an image on google drive or dropbox you can't use the direct url of your file to be an image source.
+- you need to make changes to this url in order to use it directly as an image source.
+- `goobox()` will take the url of your image file, and change it to be used directly as an image source.
+- Important : you need to check your files permessions first, and wether it's public.
+- live example: https://ybmex.csb.app/
+
+```js
+cconst goobox = (url)=>{
+
+    let dropbox_regex = /(http(s)*:\/\/)*(www\.)*(dropbox.com)/;
+    let drive_regex =/(http(s)*:\/\/)*(www\.)*(drive.google.com\/file\/d\/)/;
+
+    if(url.match(dropbox_regex)){
+        let url2 = url.replace(/(http(s)*:\/\/)*(www\.)*/, "https://dl.");
+       return url2;
+    }
+    else if(url.match(drive_regex)){
+        let url2 = url.replace(drive_regex, "");
+        let url3 = url2.match(/[\w]*\//);
+        let file_id = url3[0].replace(/\//,"")
+        let url4 = " https://drive.google.com/uc?id="+file_id
+       return url4;
+    }
+    else {
+       return console.error('Wrong URL, not a vlid drobox or google drive url');
+    }
+}
+```
+
+```js
+let url = 'https://drive.google.com/file/d/1PiCWHIwyQWrn4YxatPZDkB8EfegRIkIV/view' // can't be image src.
+
+goobox(url);
+
+//  https://drive.google.com/uc?id=1PiCWHIwyQWrn4YxatPZDkB8EfegRIkIV  // works as image src
```

Comment:
```suggestion
let driveurl = 'https://drive.google.com/file/d/1PiCWHIwyQWrn4YxatPZDkB8EfegRIkIV/view' // can't be image src.

goobox(driveurl); //  https://drive.google.com/uc?id=1PiCWHIwyQWrn4YxatPZDkB8EfegRIkIV // can be image source
```
since the google drive URL is long leaving it on a separate line is okay, but we should put the comment of the result on the same line as the call. We might also be able to omit the `// can be image source` comment?

Reply:
Is there the possibility of also getting a dropbox URL example? Also, the images need to be vetted to make sure they're a known quantity and always will be

Reply:
what you mean by vetted ??
I'll give a dropbox example.

Reply:
vetted means approved, and done in such a way that it can be made sure to never be changed to something we don't want

Reply:
@skatcat31 I've  done:
- added dropbox example.
- used more appropriate photo in my example witch is 30secs logo.
- check the photo here:  https://dl.dropbox.com/s/degru93hceo26m6/30sec.png?dl=0
- I also fixed the sandbox example